12_path_finding.asciidoc

```diff
@@ -0,0 +1,441 @@
+====
+
+==== Using MPP
+
+MPP is not something that a user will select, but rather it is a node path-finding and payment delivery strategy. The same basic steps are implemented: create a graph, select paths and the trial-and-error loop. The difference is that during path selection we must also consider how to split the payment in order to optimize delivery.
+
+In our example we can see some immediate improvements to our path finding problem that become possible with MPP. First, we can utilize the S->X channel that has known insufficient liquidity to transport 1m satoshis plus fees. By sending a smaller part along that channel, we can use paths that were previously unavailable. Second, we have the unknown liquidity of the B->R channel, which is insufficient to transport the 1m amount, but might be sufficient to transport a smaller amount.
+
+===== Splitting payments
+
+The fundamental question is how to split the payments. More specifically, what is the optimal number of splits and the optimal amounts for each split?
+
+This is an area of ongoing research, where novel strategies are emerging. Multi-path payments lead to a different algorithmic approach than single path payments, even though single-path solutions can emerge from a multi-path optimization (i.e. a single-path may be the optimal solution suggested by a multi-path path finding algorithm).
+
+If you recall, we found that the uncertainty of liquidity/balances leads to some (somewhat obvious) conclusions that we can apply in MPP path finding, namely:
+
+* Smaller payments have a higher chance of succeeding
+
+* The more channels you use, the chance of success becomes (exponentially) lower.
+
+From the first of these insights, we might conclude that splitting a large payment (e.g. 1 million satoshis) into tiny payments increases the chance that each of those smaller payments will succeed. The number of possible paths with sufficient liquidity will be greater if we send smaller amounts.
+
+To take this idea to an extreme, why not split the 1m satoshi payment into one million separate 1-satoshi parts? Well, the answer lies in our second insight: since we would be using more channels/paths to send our million single-satoshi HTLCs, our chance of success would drop exponentially.
+
+If it's not obvious, the two insights above create a "sweet spot" where we can maximize our chances of success: splitting into smaller payments but not too many splits!
+
+Quantifying this optimal balance of size/number-of-splits for a given channel graph is out of the scope of this book but it is an active area of research. Some current implementations use a very simple strategy of splitting the payment in two halves, four quarters etc.
```

Comment:
We could note from the example mentioned in my mailinglist post that this is obviously not optimal (c.f. the last paragraph of the mail: https://lists.linuxfoundation.org/pipermail/lightning-dev/2021-March/002984.html which I copy here for your convenience): 

```
One note on the question of MPP: In the last couple weeks I have been
collaborating with Christian Decker. I belief (by using the methodology
from this paper) to also have a definite solution to the question of:

How to split a payment into k parts and how many funds to allocate to each
path to increase the (multi)path success probability.

While this is is not addressed in the attached paper as we still need to
run evaluations I can already share that an equal sized split as used in
the paper (and by some implementations) is not preferable as one can easily
see from this example:
Imagine one is to deliver 100 satoshi and has to paths with 1 uncertain
channel on each path. The first of capacity 101 and  the second of 51.
Obviously trying to send 100 satoshi along the 101 capacity channel is bad.
Similarly splitting 50/50 and sending 50 Satoshi along the 51 satoshi
capacity channel is also bad. Thus a split that allocates for example 67
Satoshi to the 101 capacity and 33 to the 51 satoshi channel seems way more
reasonable. Actually 75/25 would probably be the best solution for such a
setting. And no it is only random coincident that a binary splitter would
have arrived at that split eventually (after potential miss trials)
There is way more math theory of how to actually solve the optimization
problem in the general case and how to find a split and paths that
maximizes the probability of the attempts. I cannot share these results yet
but I am pretty confident that there will be updates on that end very soon.
```